scripts/imageusagechecker.py

```diff
@@ -0,0 +1,89 @@
+import os
+import sys
+import hashlib
+import argparse
+import importlib
+import re
+from pathlib import Path
+
+
+def clean_module_path(path):
+    return (path[:-len(".py")] if path.endswith(".py") else path).replace("/", ".").replace("\\", ".")
+
+
+def build_hashes(dir):
+    locatedFiles = {}
+
+    for root, subFolders, files in os.walk(dir):
+        for file in files:
+            filename, file_extension = os.path.splitext(file)
+
+            if file_extension in ['.png', '.jpg', '.svg']:
+                filepath = os.path.join(root, file)
+                with open(filepath, 'rb') as afile:
+                    hasher = hashlib.md5()
+
+                    hasher.update(afile.read())
+                    locatedFiles[filepath] = hasher.hexdigest()
+
+    return locatedFiles
+
+
+def __main__():
+    arg_parser = argparse.ArgumentParser(description="Compares the usage of an image given two directories.")
+    arg_parser.add_argument("source", type=str, help="The image source directory")
+    arg_parser.add_argument("output", type=str, help="The image output directory")
+    arg_parser.add_argument("--exclude-file", "-e", type=str, default=None, help="Sphinx file containing document exclusions.")
+
+    args = vars(arg_parser.parse_args())
+
+    print("Directory One:", args["source"])
+    print("Directory Two:", args["output"])
+    print("Configuration Path:", args["exclude_file"])
+
+    source_hashes = build_hashes(args["source"])
+    output_hashes = build_hashes(args["output"])
+    exclude_file = args["exclude_file"]
+
+    if exclude_file is not None:
+        excluded_files = list(importlib.import_module(clean_module_path(exclude_file)).exclude_patterns)
+
+    else:
+        excluded_files = list()
+
+    err = False
+
+    print("Starting Image Scan\n")
+
+    for source_pair in source_hashes.items():
+        found = False
+        excluded = False
+
+        file_path = Path(source_pair[0])
+
+        for exc_file in excluded_files:
+            if '.png' in exc_file or '.jpg' in exc_file:
+                continue
+            else:
+                if (re.compile("docs/beta/*").search(file_path.as_posix()) is None):
+                    for build_pair in output_hashes.items():
+                        if build_pair[1] == source_pair[1]:
+                            found = True
+                else:
+                    excluded = True
+
+
+        if excluded:
+            continue
+
+        if not found:
+            print(source_pair[0] + " is not currently used!")
+            err = True
```

Comment:
```suggestion
    for file_path in source_hashes.keys():
        if not any(exc in file_path.replace('\\', '/').lower() for exc in tuple(excluded_files)):
            if source_hashes[file_path] not in output_hashes.values():
                print(file_path + " is not currently used!")
                err = True
```
Untested, something like this *should* work